bpf/lib/ratelimit.h

```diff
@@ -7,8 +7,16 @@
 #include "common.h"
 #include "bpf/helpers.h"
 
+#define RATELIMIT_USAGE_ICMPV6 1
+#define RATELIMIT_USAGE_EVENTS_MAP 2
+
 struct ratelimit_key {
-	__u32 netdev_idx;
+	__u32 usage;
+	union {
+		struct {
+			__u32 netdev_idx;
+		} icmpv6;
+	} key;
```

Comment:
Do we need to do anything wrt map migration on upgrade @dylandreimerink @joestringer @tommyp1ckles?

Reply:
The map can just be replaced without issue, the ratelimits will reset at that point, that is all. The loader code will replace the map with a new one during the upgrade, when it detects the new map definition is incompatible with the old one.

Reply:
Seems like this thread is not actionable, so resolving it.